balancer-js/src/lib/utils/poolHelper.ts

```diff
@@ -12,7 +12,7 @@ const AMP_PRECISION = 3; // number of decimals -> precision 1000
 export const parsePoolInfo = (pool: Pool) => {
   const parsedTokens = pool.tokens.map((token) => token.address);
   const parsedBalances = pool.tokens.map((token) =>
-    parseFixed(token.balance, token.decimals).toString()
```

Comment:
This actually needs to be `token.decimals`, because some tokens have decimals different than 18, such as USDC.

Reply:
yes, but the smart contract takes everything to 18, and tests work with 18 and not with token.decimals, but maybe I can study the issue further.

Reply:
You say that the join integration tests are passing with 18? Did you test using a pool that contains a token with less than 18 decimals?